tests/unit/controllers-albums.test.js

```diff
@@ -0,0 +1,390 @@
+const sinon = require('sinon');
+const album = require('../../src/controllers/album');
+const connectDb = require('../../src/services/db');
+
+describe('album controller', () => {
+  const res = {
+    status: () => {},
+  };
+  const db = {
+    query: () => {},
+    end: () => {},
+  };
+  const status = {
+    json: () => {},
+    send: () => {},
+  };
+  const req = {
+    params: {
+      albumId: 0,
+      artistId: 0,
+    },
+    body: {
+      name: 'name',
+      year: 2000,
+    },
+  };
+
+  let getDbStub;
+  let queryStub;
+  let endStub;
+  let statusStub;
+
+  beforeEach(() => {
+    getDbStub = sinon.stub(connectDb, 'getDb').resolves(db);
+    queryStub = sinon.stub(db, 'query');
+    endStub = sinon.stub(db, 'end');
+    statusStub = sinon.stub(res, 'status').returns(status);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('post', () => {
+    it('returns 201', async () => {
+      const { name, year } = req.body;
+      const { artistId } = req.params;
+      queryStub.returns([[true]]);
+
+      await album.post(req, res);
+
+      sinon.assert.calledOnce(getDbStub);
+
+      sinon.assert.calledTwice(queryStub);
+      sinon.assert.calledWith(queryStub, `SELECT * FROM Artist WHERE id = ?`, [
+        artistId,
+      ]);
+      sinon.assert.calledWith(
+        queryStub,
+        `INSERT INTO Album (name, year, artistId) VALUES (?, ?, ?)`,
+        [name, year, artistId]
+      );
+
+      sinon.assert.calledOnce(statusStub);
+      sinon.assert.calledWith(statusStub, 201);
+
+      sinon.assert.calledOnce(endStub);
+    });
+
+    it('returns 404', async () => {
+      const { artistId } = req.params;
+      queryStub.returns([[false]]);
+
+      await album.post(req, res);
+
+      sinon.assert.calledOnce(getDbStub);
+
+      sinon.assert.calledOnce(queryStub);
+      sinon.assert.calledWith(queryStub, `SELECT * FROM Artist WHERE id = ?`, [
+        artistId,
+      ]);
+
+      sinon.assert.calledOnce(statusStub);
+      sinon.assert.calledWith(statusStub, 404);
+
+      sinon.assert.calledOnce(endStub);
```

Comment:
I see what youre going for here separating each stub into blocks for their assertions, its not really necessary given how few there are and how easy this is to read. Personally Id say remove the blank lines separating them